### PR TITLE
Decouple databroker search and summary UI from bluesky-browser

### PR DESCRIPTION
**Pull request by dylanmcreynolds**

### CODE REVIEW

File: xicam/gui/bluesky/top_utils.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from traitlets import TraitType
+from traitlets.config.loader import (PyFileConfigLoader, ConfigFileNotFound,
+                                     Config)
+
+CONFIG_FILE_NAME = 'bluesky_browser_config.py'
```
Comment (danielballan) [inline xicam/gui/bluesky/top_utils.py]: This should probably be updated. In the future we can have a larger discussion about configuration management, not sure if that should hold up this PR.

Comment (danielballan) [inline xicam/gui/bluesky/top_utils.py:5]: This should probably be updated. In the future we can have a larger discussion about configuration management, not sure if that should hold up this PR.

File: xicam/gui/bluesky/search.py [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+    QVBoxLayout,
+    QWidget,
+    QTableView,
+    )
+from .utils import ConfigurableQObject
+from .top_utils import load_config, Callable
+
+
+MAX_SEARCH_RESULTS = 100  # TODO Use fetchMore instead of a hard limit.
+log = logging.getLogger('bluesky_browser')
```
Comment (danielballan) [inline xicam/gui/bluesky/search.py:36]: This should probably be changed to `'search_widget'` or, if Xi-cam already has a convention for logging scope, `'SOMETHING.search_widget'`.

File: setup.py [hunk 3]
```diff
@@ -82,7 +82,7 @@
     # your project is installed. For an analysis of "install_requires" vs pip's
     # requirements files see:
     # https://packaging.python.org/en/latest/requirements.html
-    install_requires=["qtpy", "pathlib", "pyqtgraph", "qdarkstyle", "qtmodern", "qtconsole", "xicam.plugins", "xicam.core", "bluesky-browser"]
+    install_requires=["qtpy", "pathlib", "pyqtgraph", "qdarkstyle", "qtmodern", "qtconsole", "xicam.plugins", "xicam.core"]
```
Comment (danielballan) [inline setup.py:85]: :tada:

File: xicam/gui/bluesky/utils.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+            if next_index < len(mro):
+                mro[next_index].__init__(self, *args, **kw)
+
+    return SuperQClass
+
+
+SuperQObject = superQ(QObject)
+ConfigurableQObject = MetaQObjectHasTraits(
+    'NewBase', (Configurable, SuperQObject), {})
+ConfigurableQTabWidget = MetaQObjectHasTraits(
```
Comment (danielballan) [inline xicam/gui/bluesky/utils.py:148]: This is no longer used.

File: xicam/gui/bluesky/utils.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    'NewBase', (Configurable, QTabWidget, SuperQObject), {})
+ConfigurableMoveableTabContainer = MetaQObjectHasTraits(
```
Comment (danielballan) [inline xicam/gui/bluesky/utils.py:150]: This is no longer used.

File: xicam/gui/bluesky/utils.py [hunk 6]
```diff
@@ -0,0 +1,151 @@
+import inspect
+
+from PyQt5.QtGui import QCursor, QDrag, QPixmap, QRegion
+from PyQt5.QtWidgets import QWidget, QTabWidget
+from PyQt5.QtCore import Qt, QMimeData, QObject, QPoint
+from traitlets import HasTraits, TraitType
+from traitlets.config import Configurable
+
+
+class MoveableTabWidget(QTabWidget):
```
Comment (danielballan) [inline xicam/gui/bluesky/utils.py:10]: This is no longer used.

File: xicam/gui/bluesky/utils.py [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+        if counter == 0:
+            self.addTab(e.source().parentWidget().widget(self.parent().tab_index),
+                        e.source().tabText(self.parent().tab_index))
+        else:
+            self.insertTab(counter + 1,
+                           e.source().parentWidget().widget(self.parent().tab_index),
+                           e.source().tabText(self.parent().tab_index))
+
+
+class MoveableTabContainer(QWidget):
```
Comment (danielballan) [inline xicam/gui/bluesky/utils.py:70]: This is no longer used.

**Review by danielballan — COMMENTED** [review]
Thanks for putting this together so quickly. I left some suggestions, just things we can rename and remove if we like. Nothing critical.

**Review by danielballan — CHANGES_REQUESTED** [review]
Consider #35 as an alternative way of getting to the same code.

**Comment by ronpandolfi** [comment]
Closing in favor of https://github.com/synchrotrons/Xi-cam.gui/pull/35

**Pull request closed without merging**